src/compiler/factoryPublic.ts

```diff
@@ -1412,9 +1412,11 @@ namespace ts {
     }
 
     export function updateBinary(node: BinaryExpression, left: Expression, right: Expression, operator?: BinaryOperator | BinaryOperatorToken) {
+        const nextOperator = operator ?? node.operatorToken;

```

Comment:
If you're going to do it this way, you can use an optional parameter instead

```suggestion
    export function updateBinary(node: BinaryExpression, left: Expression, right: Expression, operator?: BinaryOperator | BinaryOperatorToken = node.operator) {
```